pkg/controller/jobframework/base_webhook.go

```diff
@@ -55,7 +59,11 @@ func (w *BaseWebhook) Default(ctx context.Context, obj runtime.Object) error {
 	job := w.FromObject(obj)
 	log := ctrl.LoggerFrom(ctx)
 	log.V(5).Info("Applying defaults", "job", klog.KObj(job.Object()))
-	ApplyDefaultForSuspend(job, w.ManageJobsWithoutQueueName)
+	if features.Enabled(features.KueueDefaulting) && w.Queues.DefaultLocalQueue(job.Object().GetNamespace()) != "" {
```

Comment:
Maybe I miss the discussion but I read this change to mean:

Once KueueDefaulting is true, we will no longer use w.ManageJobsWithoutQueueName? 

Should we consider deprecation of w.ManageJobsWithoutQueueName?

Reply:
In k/k, we always take feature gates to eventually mean that they will be true so we should think through when we want to remove this feature gate and always opt in for this feature.

Reply:
Good point. With KueueDefaulting ManageJobsWithoutQueueName doesn't make sense, so in case the feature go to GA, ManageJobsWithoutQueueName should be deprecated.

Reply:
However, if the default queue doesn't exist, ManageJobsWithoutQueueName is still useful.

Reply:
^^ agree here. I was thinking of this as just a QoL improvement especially for cases where users have access to only 1 LocalQueue. It allows them to ignore any underlying Kueue details and submit their jobs as normal